frontend/src/static/js/v_summary.js

```diff
@@ -358,7 +370,10 @@ window.vSummary = {
       // array of array, sorted by repo
       const full = [];
 
-      this.repos.forEach((repo) => {
+      // create deep clone of this.repos if merging groups
+      const groups = this.isMergeGroup ? JSON.parse(JSON.stringify(this.repos)) : this.repos;
```

Comment:
If we do not watch `repos` for now, do we still need this deep clone?

Reply:
Yes we do as we do not want to affect the original `repos` when we are merging here:
https://github.com/reposense/RepoSense/blob/65c1766c8c5459984c0437f3b1f754af46bef3d3/frontend/src/static/js/v_summary.js#L432-L434

Reply:
OK